pyexasol/formatter.py

```diff
@@ -51,32 +67,91 @@ def convert_field(self, value, conversion):
 
     @classmethod
     def escape(cls, val):
+        """
+        Takes a raw value and converts it into an and escaped string.
+
+        Args:
+            val: Value to be escaped.
+
+        Returns:
+            A string where all single quotes ``'`` have been replaced
+            with two single quotes ``''``.
+        """
         return str(val).replace("'", "''")
 
     @classmethod
     def escape_ident(cls, val):
+        """
+        Takes a raw value and converts it into an and escaped string.
+
+        Args:
+            val: Value to be escaped.
+
+        Returns:
+            A string where all double quotes ``"`` have been replaced 
+            with two double quotes ``""``.
+        """
         return str(val).replace('"', '""')
 
     @classmethod
     def escape_like(cls, val):
+        """
+        Escape LIKE-patterns.
+
+        Args:
+            val: Value to be escaped.
+
+        Returns:
+            A string where all double quotes ``\\`` have been replaced 
+            with ``\\\\``, where ``%`` have been replaced with ``\%``,
+            where ``_`` have been replaced with ``\_``.
+        """
         return cls.escape(val).replace('\\', '\\\\').replace('%', r'\%').replace('_', r'\_')
 
     @classmethod
     def quote(cls, val):
+        """
+        Escapes a string using :meth:`pyexasol.ExaFormatter.escape` and wraps it in single quotes ``'``.
+        """
         if val is None:
             return 'NULL'
 
         return f"'{cls.escape(val)}'"
 
     @classmethod
     def quote_ident(cls, val):
+        """
+        Escapes a string one or multiple values using :meth:`pyexasol.ExaFormatter.excape_ident` and wraps it in double quotes ``"``.
+
+        Args:
+            val (str or tuple): Raw identifier(s) to be escaped.
```

Comment:
Actually val does not need to be a str because `escape_ident` calls str(...) on the object.